src/app/features/post-overview/components/post-detailview/post-detailview.component.ts

```diff
@@ -26,4 +40,34 @@ export class PostDetailviewComponent implements OnInit, OnDestroy {
   ngOnDestroy(): void {
     this.routeSub.unsubscribe();
   }
+
+  onDelete(): void {
+    this.routeSub = this.currentRoute.params.subscribe((params: Params) => {
+      this.service
+        .removePostFrom(params.blogId, params.postId)
+        .subscribe((_) =>
+          this.service.navigateTo(`/home/blogs/${params.blogId}`)
+        );
+    });
+  }
+
+  onEdit(post: Post): void {
+    this.routeSub = this.currentRoute.params.subscribe((params: Params) => {
+      const dialogRef = this.service.openDialog(post);
+      this.dialogSub = dialogRef
+        .afterClosed()
+        .subscribe((body: PostRequestBody) => {
+          if (body) {
+            this.editPostSub = this.service
+              .editPost(params.blogId, body)
+              .subscribe(
+                (editedPost: Post) => this.ngOnInit()
+                /* this.service.navigateTo(
+                  `/home/blogs/${params.blogId}/posts/${params.postId}/detail`
+                ) */
+              );
+          }
+        });
+    });
```

Comment:
hier Zeile 55 bzw. 71 entfernen, also die routeSub und statt params.blogId post.blog.id nutzen den du aus dem template übergeben bekommen kannst :)

In Zeile 64 kannst du dann statt ngOnInit() das hier einfügen:
```ts
(editedPost: Post) => this.post$ = this.service.getPostById(post.id)
```
Weil ngOnInit ist eig eine LifeCycle methode, die nur von  Angular selbst getriggered werden sollte, anstatt sie aktiv von unserer Seite aus aufzurufen

Sonst noch den Comment dadrunter weg :)